docs/GettingStartedDocs/VisualStudioLinux.md

```diff
@@ -0,0 +1,147 @@
+   app template.  In VS2017, this is called "Console Application (Linux)", and in VS2019,
+   it is called "Console App" (note: NOT the "Console App (.NET Core)") with the Linux
+   keyword.  Give the project a name, LinuxApp for example.  This will create a "Hello World" console application.  
+   Alternatively, if you already have such a Linux application using a Visual Studio project
+   file (.vcxproj file), you can start from your existing application.
+2. Configure the application project to use your Linux build environment, by right clicking
+   on the project in the Solution Explorer and selecting Properties -> Configuration
+   Properties -> General -> Remote Build Machine, and explicitly set it to the build machine
+   you configured in the Connection Manager.  (Due to a current Visual Studio bug, this step
+   is required even if the correct value is shown by default. In other words, make sure the
+   connection is shown in **bold**.)
+3. If using Visual Studio 2019, also update Configuration Properties -> Debugging -> Remote
+   Debug Machine to your build machine, again due to a current Visual Studio 2019 bug.
+   (This step is not needed on Visual Studio 2017.)  At this point, you should be able to
+   build and debug the Hello World application. For further discussion, see the
+   [Linux debugging walkthrough](https://docs.microsoft.com/en-us/cpp/linux/deploy-run-and-debug-your-linux-project?view=vs-2019).
+4. Create an enclave library project by right clicking on the solution in the Solution Explorer
+   and selecting Add -> New Project -> Open Enclave TEE Project (Linux).   Give it a name,
+   LinuxEnclave for example.  This will create a sample enclave with an ecall\_DoWorkInEnclave()
+   method exposed to applications, that will simply call an ocall\_DoWorkInHost() method that
+   will be implemented in the application.   In this walkthrough, we'll leave this project
+   as is for now, but afterwards you can modify it as you like.
+5. Configure the enclave project to use your Linux build environment, as you did in step 2.
+   At this point, the enclave would build, but cannot be run as the application doesn't
+   invoke it yet.
+6. Import the enclave into your application project, by right clicking on the application
+   project in the Solution Explorer and selecting Open Enclave Configuration -> Import Enclave,
+   then navigate to and select the EDL file (_YourEnclaveProjectName_.edl) in your enclave project.
+   This step will modify your application project settings and add some additional files to it,
+   including a C file named _YourEnclaveProjectName_\_host.c.  This C file contains a
+   sample\_enclave\_call() method that will load and call
+   ecall\_DoWorkInEnclave(), and also contains a sample implementation of a ocall\_DoWorkInHost()
+   method that just prints a message when called.  Although the app could be compiled and run
+   at this point, sample\_enclave\_call() is still not called from anywhere.
```

Comment:
On my environment, every time I do a build, all the source files in the project are recompiled unlike a normal VS project where only the modified files are recompiled.

I don't know if this is only on my Windows machine or not. I am using a Linux ACC VM for remote building/running.

Reply:
As far as I can tell, the Linux support for VS always does this.